### PR TITLE
Refactored custom logger 

### PR DESCRIPTION
**Pull request by PawelK2012**
Refactored custom logger in order to add capacity to handle custom attributes.

### CODE REVIEW

File: logger/custom_logger.go [hunk 1]
```diff
@@ -27,19 +28,161 @@ const (
...
-func (h *CustomHandler) Handle(c context.Context, r slog.Record) error {
-	timestamp := time.Now().Format(time.RFC3339Nano)
+type Options struct {
+	// Level reports the minimum level to log.
+	// Levels with lower levels are discarded.
+	// If nil, the Handler uses [slog.LevelInfo].
+	Level slog.Leveler
+}
+
+func NewCustomHandler(out io.Writer, opts *Options) *CustomHandler {
```
Comment (vladoatanasov) [inline logger/custom_logger.go:46]: can you add a comment for this exported function? Something like:
```
// NewCustomHandler returns a new custom handler instance. If no log level is provided in opts, LevelInfo is used by default.
```

File: logger/custom_logger.go [hunk 2]
```diff
@@ -27,19 +28,161 @@ const (
...
+	if h.opts.Level == nil {
+		h.opts.Level = slog.LevelInfo
+	}
+	return h
+}
+
+// Handle is a custom handler for logging records.
+func (ch *CustomHandler) Handle(ctx context.Context, r slog.Record) error {
+	indentLevel := 0
+	bufp := allocBuf()
```
Comment (vladoatanasov) [inline logger/custom_logger.go:60]: is this pointer needed? I don't see it used anywhere in the function. You assign the value that it's pointing to to buf and that's what you are using throughout the rest of the function.

File: logger/custom_logger.go [hunk 3]
```diff
@@ -27,19 +28,161 @@ const (
...
+		})
+	}
+
+	// adding \n at the end for better formatting
+	buf = append(buf, "\n"...)
+	ch.mu.Lock()
+	defer ch.mu.Unlock()
+	_, err := ch.out.Write(buf)
+	if err != nil {
+		fmt.Println("write out error ", err)
```
Comment (vladoatanasov) [inline logger/custom_logger.go:84]: this print statement seems redundant and can be removed

File: logger/custom_logger.go (hunk 2 above)
Comment (PawelK2012) [reply logger/custom_logger.go:60]: `buffp` stands for buf Pool. See what `allocBuf()` returns at line 219.

File: logger/custom_logger.go [hunk 4]
```diff
@@ -27,19 +28,161 @@ const (
...
+	return h
+}
+
+// Handle is a custom handler for logging records.
+func (ch *CustomHandler) Handle(ctx context.Context, r slog.Record) error {
+	indentLevel := 0
+	bufp := allocBuf()
+	buf := *bufp
+	defer func() {
+		*bufp = buf
```
[the remark this answers is not in the document]
Comment (PawelK2012) [reply logger/custom_logger.go:65]: See [comment](https://github.com/IBM/opentalaria/pull/7#discussion_r1832782997) below.

**Review by vladoatanasov — APPROVED** [review]
Looks good!

**Pull request merged**